yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtText.kt

```diff
@@ -3,34 +3,146 @@ package dev.yorkie.document.crdt
 import dev.yorkie.document.time.ActorID
 import dev.yorkie.document.time.TimeTicket
 
+/**
+ * [CrdtText] is a custom CRDT data type to represent the contents of text editors.
+ */
 internal data class CrdtText(
+    val rgaTreeSplit: RgaTreeSplit<TextValue>,
     override val createdAt: TimeTicket,
-    override var _movedAt: TimeTicket?,
-    override var _removedAt: TimeTicket?,
+    override var _movedAt: TimeTicket? = null,
+    override var _removedAt: TimeTicket? = null,
 ) : CrdtElement() {
+    private val selectionMap = mutableMapOf<ActorID, Selection>()
+
+    private var onChangesHandler: ((List<TextChange>) -> Unit)? = null
+
+    private var remoteChangeLock: Boolean = false
```

Comment:
we might have to mark this `volatile` for thread-safety.
though it could still not be enough depending on the usages.